cmd/appliance/upgrade/complete.go

```diff
@@ -810,6 +810,23 @@ func upgradeCompleteRun(cmd *cobra.Command, args []string, opts *upgradeComplete
 		}
 	}
 
+	// Trigger ZTP version update if needed
+	// From v18 and up
+	// This step is not fatal, so we only log errors here
+	if opts.Config.Version >= 18 {
+		ZTPStatus, err := a.ZTPStatus(ctx)
+		if err != nil {
+			log.WithError(err).Warn("failed to get ZTP registered status")
+		}
+		if ZTPRegistered, ok := ZTPStatus.GetRegisteredOk(); err == nil && ok {
```

Comment:
Shouldn't you skip this part if ZTPStatus is nil?
Just a style thing but is it normal to name variables with leading uppercase like this? Looked like a type to me at first.